src/Microsoft.ML.TimeSeries/SsaAnomalyDetectionBase.cs

```diff
@@ -201,6 +209,47 @@ public sealed class State : AnomalyDetectionStateBase
             private SequenceModelerBase<Single, Single> _model;
             private SsaAnomalyDetectionBase _parentAnomalyDetector;
 
+            public State()
+            {
+
+            }
+
+            public State(ModelLoadContext ctx) : base(ctx)
+            {
+                WindowedBuffer = new FixedSizeQueue<float>(
+                    ctx.Reader.ReadInt32(), ctx.Reader.ReadInt32(), ctx.Reader.ReadSingleArray());
+
+                InitialWindowedBuffer = new FixedSizeQueue<float>(
+                    ctx.Reader.ReadInt32(), ctx.Reader.ReadInt32(), ctx.Reader.ReadSingleArray());
+            }
+
+            public override void Save(ModelSaveContext ctx)
+            {
+                base.Save(ctx);
+
+                ctx.Writer.Write(WindowedBuffer.Capacity);
+                ctx.Writer.Write(WindowedBuffer.StartIndex);
+                ctx.Writer.WriteSingleArray(WindowedBuffer.Buffer);
+
```

Comment:
So, just to be clear what you'd do here once you get rid of those public accessors to the guts of the buffer, you'd write `WindowedBuffer.Count`, then write each element of that array.

If I were you I'd write a little utility serializing/deserializing these windowed buffers.

Reply:
What about start index? This is a circular buffer right?

---
In reply to: [236476800](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236476800) [](ancestors = 236476800)

Reply:
Yes, of course. And what do you think the difference between a circular buffer is with buffer `{1,2,3}` and start index `0`, and one with a buffer `{2, 3, 1}` and start index 2?

---
In reply to: [236477016](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236477016) [](ancestors = 236477016,236476800)

Reply:
feeling a little stupid now...may be lack of sleep, thanks! 

---
In reply to: [236481618](https://github.com/dotnet/machinelearning/pull/1727#discussion_r236481618) [](ancestors = 236481618,236477016,236476800)